react_juce/core/EcmascriptEngine_Hermes.cpp

```diff
@@ -0,0 +1,522 @@
+            };
+        }
+
+        juce::var jsiObjectToVarObject(jsi::HostObject &v, jsi::Runtime &runtime)
+        {
+            juce::DynamicObject::Ptr varObj = new juce::DynamicObject();
+
+            auto props = v.getPropertyNames(runtime);
+            for (auto &propID : props)
+            {
+                jsi::Value property = v.get(runtime, propID);
+                varObj->setProperty(juce::String(propID.utf8(runtime)), jsiValueToVar(property, runtime));
+            }
+
+            return varObj.get();
+        }
+
+        juce::var jsiValueToVar(const jsi::Value &v, jsi::Runtime &runtime)
+        {
+            if (v.isBool())
+                return v.getBool();
+
+            if (v.isNumber())
+                return v.getNumber();
+
+            if (v.isString())
+                return juce::String(v.getString(runtime).utf8(runtime));
+
+            if (v.isUndefined())
+                return juce::var::undefined();
+
+            if (v.isNull())
+                return juce::var();
+
+            if (v.isObject())
+            {
+                auto obj = v.getObject(runtime);
+
+                if (obj.isArray(runtime))
+                    return jsiArrayToVarArray(obj.getArray(runtime), runtime);
+
+                if (obj.isFunction(runtime))
+                    return jsiMethodToVarMethod(obj.asFunction(runtime), runtime);
+
+                if (obj.isHostObject(runtime))
+                {
+                    auto hostObjectPtr = obj.getHostObject(runtime);
+                    return jsiObjectToVarObject(*hostObjectPtr, runtime);
+                }
+
+                if (obj.isArrayBuffer(runtime))
+                {
+                    jsi::ArrayBuffer arrayBuffer = obj.getArrayBuffer(runtime);
+                    juce::MemoryBlock memBlock(static_cast<const void*>(arrayBuffer.data(runtime)),
+                                                                        arrayBuffer.size(runtime));
+
+                    return juce::var(memBlock);
+                }
+            }
+
+            return {};
+        }
+
+        //==============================================================================
+        jsi::Object varObjectToJSIObject(juce::DynamicObject *v, jsi::Runtime &runtime)
+        {
+            jassert(v);
+
+            jsi::Object jsiObj(runtime);
+            for (auto &prop : v->getProperties())
+            {
+                jsiObj.setProperty(runtime, prop.name.getCharPointer(), varToJSIValue(prop.value, runtime));
+            }
+
+            return jsiObj;
+        }
+
+        jsi::Function varMethodToJSIMethod(juce::var::NativeFunction &v, jsi::Runtime &runtime)
+        {
+            return jsi::Function::createFromHostFunction(
+                runtime,
+                jsi::PropNameID::forAscii(runtime, ""),
+                0,
+                [v = v](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count)
```

Comment:
Love how easy that is. Believe we could also `v = std::move(v)` here if we want

Reply:
Yeah move here should be good. Will check.

Reply:
Thinking about it, is move actually right here? What if user has pushed a `DynamicObject` through to JS? We could be attempting to move an objects method prop? i.e. we have to do the copy somewhere. Just happensere based on the code that it's here.

Reply:
See here: 

```
        jsi::Object varObjectToJSIObject(juce::DynamicObject *v, jsi::Runtime &runtime)
        {
            jassert(v);

            jsi::Object jsiObj(runtime);
            for (auto &prop : v->getProperties())
            {
                jsiObj.setProperty(runtime, prop.name.getCharPointer(), varToJSIValue(prop.value, runtime));
            }

            return jsiObj;
        }
```

`varToJSIValue` in the above block will call through to `varMethodToJSIMethod` for the object props.

Reply:
Again, in case lost in the weeds. What dya reckon to this one based on above explanation @nick-thompson ?

Reply:
Ahhh, right. Ok, let's avoid the move